File: patch_worklet2.js
<<<<<<< SEARCH
                const processingTime = (currentTime - startTime) * 1000; // Convert to ms
                this.totalProcessTime += processingTime;
                this.processCount++;

                // Send result to main thread
                this.port.postMessage({
                    type: 'pitch',
                    pitch: result.pitch,
                    confidence: result.confidence,
                    timestamp: currentTime,
                    latency: processingTime,
=======
                const processingTime = (globalThis.currentTime - startTime) * 1000; // Convert to ms
                this.totalProcessTime += processingTime;
                this.processCount++;

                // Send result to main thread
                this.port.postMessage({
                    type: 'pitch',
                    pitch: result.pitch,
                    confidence: result.confidence,
                    timestamp: globalThis.currentTime,
                    latency: processingTime,
>>>>>>> REPLACE

File: patch_worklet2.py
with open("src/audio/PitchWorklet.js", "r") as f:
    content = f.read()

content = content.replace('const startTime = currentFrame / sampleRate;', 'const startTime = globalThis.currentTime;')
content = content.replace('const processingTime = ((currentFrame / sampleRate) - startTime) * 1000;', 'const processingTime = (globalThis.currentTime - startTime) * 1000;')
content = content.replace('timestamp: currentFrame / sampleRate,', 'timestamp: globalThis.currentTime,')

with open("src/audio/PitchWorklet.js", "w") as f:
    f.write(content)
